tests/src/LuaTests.cs

```diff
@@ -983,6 +983,71 @@ public void SetGlobalObject()
             }
         }
         ///*
+        // * Tests setting of a global variable to a CLR object and checking if the Globals correctly registered everything.
+        // */
+        [Test]
+        public void SetGlobalObjectLuaGlobalsListIsCorrect()
+        {
+            using (Lua lua = new Lua())
+            {
+                lua["netobj"] = new TestTypes.GlobalsTestClass();
+
+                var globals = lua.Globals.ToList();
+                Assert.AreEqual(globals.Count, 4);
+                Assert.Contains("netobj.Property1", globals);
+                Assert.Contains("netobj.Property2", globals);
+                Assert.Contains("netobj:Method1()", globals);
+                Assert.Contains("netobj:Method3(", globals);
+            }
+        }
+        ///*
+        // * Tests setting of a global variable to a CLR object value and then re assigning it to a non CLR object.
+        // */
+        [Test]
+        public void SetGlobalObjectAndReasignToNonCLR()
+        {
+            using (Lua lua = new Lua())
+            {
+                lua["netobj"] = new TestTypes.GlobalsTestClass();
+                lua["netobj"] = 4;
+                var globals = lua.Globals.Where(x => x.StartsWith("netobj")).ToList();
+                Assert.AreEqual(1, globals.Count);
+                Assert.Contains("netobj", globals);
+            }
+        }
+        ///*
+        // * Tests setting of a global variable to a CLR object value and then re assigning it to a null value.
+        // */
+        [Test]
+        public void SetGlobalObjectAndReasignToNull()
+        {
+            using (Lua lua = new Lua())
+            {
+                lua["netobj"] = new TestTypes.GlobalsTestClass();
+                lua["netobj"] = null;
+                var globals = lua.Globals.Where(x => x.StartsWith("netobj")).ToList();
+                Assert.AreEqual(0, globals.Count);
+            }
+        }
+        ///*
+        // * Tests setting of a global variable to a CLR object value and then re assigning it to another CLR object of another type.
+        // */
+        [Test]
+        public void SetGlobalObjectAndReasignToOtherCLR()
+        {
+            using (Lua lua = new Lua())
+            {
+                lua["netobj"] = new TestTypes.TestClass();
+                lua["netobj"] = new TestTypes.GlobalsTestClass();
+                var globals = lua.Globals.Where(x => x.StartsWith("netobj")).ToList();
+                Assert.AreEqual(4, globals.Count);
+                Assert.Contains("netobj.Property1", globals);
+                Assert.Contains("netobj.Property2", globals);
+                Assert.Contains("netobj:Method1()", globals);
+                Assert.Contains("netobj:Method3(", globals);
```

Comment:
You might need to change to `Assert.IsTrue(global.Contains("netobj.Property1", "#1);`

NUnit on iOS and tvOS have no Assert.Contains

```
(CoreCompile target) -> 
  /Users/runner/work/1/s/tests/src/LuaTests.cs(997,24): error CS0117: 'Assert' does not contain a definition for 'Contains' [/Users/runner/work/1/s/tests/build/xamarinios/NLuaTest.XamariniOS.csproj]
  /Users/runner/work/1/s/tests/src/LuaTests.cs(998,24): error CS0117: 'Assert' does not contain a definition for 'Contains' [/Users/runner/work/1/s/tests/build/xamarinios/NLuaTest.XamariniOS.csproj]
  /Users/runner/work/1/s/tests/src/LuaTests.cs(999,24): error CS0117: 'Assert' does not contain a definition for 'Contains' [/Users/runner/work/1/s/tests/build/xamarinios/NLuaTest.XamariniOS.csproj]
  /Users/runner/work/1/s/tests/src/LuaTests.cs(1000,24): error CS0117: 'Assert' does not contain a definition for 'Contains' [/Users/runner/work/1/s/tests/build/xamarinios/NLuaTest.XamariniOS.csproj]
  /Users/runner/work/1/s/tests/src/LuaTests.cs(1015,24): error CS0117: 'Assert' does not contain a definition for 'Contains' [/Users/runner/work/1/s/tests/build/xamarinios/NLuaTest.XamariniOS.csproj]
  /Users/runner/work/1/s/tests/src/LuaTests.cs(1044,24): error CS0117: 'Assert' does not contain a definition for 'Contains' [/Users/runner/work/1/s/tests/build/xamarinios/NLuaTest.XamariniOS.csproj]
  /Users/runner/work/1/s/tests/src/LuaTests.cs(1045,24): error CS0117: 'Assert' does not contain a definition for 'Contains' [/Users/runner/work/1/s/tests/build/xamarinios/NLuaTest.XamariniOS.csproj]
  /Users/runner/work/1/s/tests/src/LuaTests.cs(1046,24): error CS0117: 'Assert' does not contain a definition for 'Contains' [/Users/runner/work/1/s/tests/build/xamarinios/NLuaTest.XamariniOS.csproj]
  /Users/runner/work/1/s/tests/src/LuaTests.cs(1047,24): error CS0117: 'Assert' does not contain a definition for 'Contains' [/Users/runner/work/1/s/tests/build/xamarinios/NLuaTest.XamariniOS.csproj]

```

Reply:
Ah good one will update asap. Can I test for these things om a Windows machine or would I need a MacOS machine for this?

Reply:
Fixed